bridge/irc_manager.go

```diff
@@ -333,14 +344,16 @@ func (m *IRCManager) SendMessage(channel string, msg *DiscordMessage) {
 
 	// Person is appearing offline (or the bridge is running in Simple Mode)
 	if !ok {
-		length := len(msg.Author.Username)
+		// length := len(msg.Author.Username)
 		for _, line := range strings.Split(content, "\n") {
-			m.bridge.ircListener.Privmsg(channel, fmt.Sprintf(
-				"<%s#%s> %s",
-				msg.Author.Username[:1]+"\u200B"+msg.Author.Username[1:length],
-				msg.Author.Discriminator,
-				line,
-			))
+			// m.bridge.ircListener.Privmsg(channel, fmt.Sprintf(
+			// 	"<%s#%s> %s",
+			// 	msg.Author.Username[:1]+"\u200B"+msg.Author.Username[1:length],
+			// 	msg.Author.Discriminator,
+			// 	line,
+			// ))
+
+			m.bridge.ircListener.Privmsg(channel, m.formatIRCMessage(msg, line))
```

Comment:
Can we make `m.formatIRCMessage` return a slice (so that it only needs to be passed `msg`)?

Reply:
You mean move  `m.bridge.ircListener.Privmsg(channel, m.formatIRCMessage(msg, line))` out of the "for range" and have it just take `msg`? Probably better to do it that way actually, instead of formatting per channel. Could you explain what you mean by return a slice however?

Reply:
Wait, rereading the code, but basically the for range wrapping this splits the line on '\n' and that's why I made it take a line argument. The function is to format lines, not to split lines out and format them. That said the formatting works on things besides lines anyway so still moving that outside of the for range and letting it then be split on '\n' works too I would think.

Reply:
Thinking through this more, it wouldn't work, `formatIRCMessage` would need to be aware of new lines '\n' and apply each line as content.

Say the format is `"\x02${USER}\x02 ${CONTENT}"` - if we treat the entire `msg.Content` as the singular content then only the first line will have the `"\x02${USER}\x02"` part and the rest of the lines will have no prefix of the user.

That or I'm misunderstanding what is wanted to be done...

The reason `formatIRCMessage` needs to remain unaware that it may be handling more than one line is if I add support for including the referenced content in the message (for edit/reply) then this will also be applying for IRCConnections when pushing into their `messages` channel.